modules/core/src/main/scala/lucuma/odb/api/repo/ObservationRepo.scala

```diff
@@ -230,55 +181,79 @@ object ObservationRepo {
             case Assign(tid) => tid.asRight[Asterism.Id].some
           }
 
+        val updateConstraintSet: Option[ConstraintSet.Id] => Option[ConstraintSet.Id] = in =>
+          constraintSet match {
+            case Ignore        => in
+            case Unassign      => None
+            case Assign(csid) => csid.some
+          }
+
         val updatePointing = updateAsterism andThen updateTarget
+        
+        // Don't send a change event for the old id if the id Input is Ignore or the value is unchanged.
+        // Also don't send an event for the new id (if Assign) if it is always the same as the old id.
+        def ids4Event[A: Eq](oa: Option[A], ia: Input[A]): List[A] = (oa, ia) match {
+          case (_, Ignore)                => Nil
+          case (Some(a), Unassign)        => List(a)
+          case (None, Unassign)           => Nil
+          case (Some(aOld), Assign(aNew)) => if (aOld === aNew) Nil else List(aOld, aNew)
+          case (None, Assign(aNew))       => List(aNew)
+        }
```

Comment:
Thanks, I think I was generating extra events before.

Reply:
Yeah, it was generating events for the current pointing on an edit event even if it wasn't being changed.